egs/wsj/s5/steps/data/reverberate_data_dir.py

```diff
@@ -0,0 +1,299 @@
+# The noise list would have the following format:
+# --noise-id <string,compulsary> --noise-type <choices = (isotropic, point source),compulsary> --bg-fg-type <choices=(background|foreground), default=background> --rir-file <str, compulsary if isotropic, should not be specified if point-source> < location=(support Kaldi IO strings) >
+def CorruptWav(wav_scp, durations, output_dir, room_list, noise_list, snr_string, num_replica, prefix, speech_rvb_probability, noise_adding_probability, max_noises_added):
+    rooms = list_cyclic_iterator(room_list, random_seed = 1)
+    noises = None
+    if len(noise_list) > 0:
+        noises = list_cyclic_iterator(noise_list, random_seed = 1)
+    snrs = list_cyclic_iterator(snr_string.split(':'))
+    command_list = []
+    for i in range(num_replica):
+        keys = wav_scp.keys()
+        keys.sort()
+        for wav_id in keys:
+            wav_pipe = wav_scp[wav_id]
+            wav_dur = durations[wav_id]
+            if prefix is not None:
+                wav_id = prefix + str(i) + "_" + wav_id
+            command = "{0} {1} wav-reverberate".format(wav_id, wav_pipe)
+
+            # pick the room
+            room = rooms.next()
+            command_opts = ""
+            noises_added = []
+            snrs_added = []
+            start_times_added = []
+            if random.random() < speech_rvb_probability:
+                # pick the RIR to reverberate the speech
+                speech_rir = room['rir_list'][random.randint(0,len(room['rir_list'])-1)]
+                command_opts += "--impulse-response={0} ".format(speech_rir.rir_file_location)
+                # add the corresponding isotropic noise if there is any
+                if len(speech_rir.iso_noise_list) > 0:
+                    isotropic_noise = speech_rir.iso_noise_list[random.randint(0,len(speech_rir.iso_noise_list)-1)]
+                    noises_added.append("{0}".format(isotropic_noise.noise_file_location))
+                    snrs_added.append("{0}".format(snrs.next()))
+                    start_times_added.append(round(random.random() * wav_dur, 2))
+
+            if noises is not None and random.random() < noise_adding_probability:
+                for k in range(random.randint(1, max_noises_added)):
+                    # pick the RIR to reverberate the point-source noise
+                    noise = noises.next()
+                    noise_rir = room['rir_list'][random.randint(0,len(room['rir_list'])-1)]
+                    start_times_added.append(round(random.random() * wav_dur, 2))
+                    noises_added.append("\"wav-reverberate --duration={2} --impulse-response={1} {0} - |\" ".format(noise.noise_file_location, noise_rir.rir_file_location, round(random.random()*(wav_dur-start_times_added[-1]), 2)))
+                    snrs_added.append("{0}".format(snrs.next()))
```

Comment:
Yes.

On Thu, Apr 21, 2016 at 4:44 AM, tomkocse <notifications@github.com> wrote:

> In egs/wsj/s5/steps/data/reverberate_data_dir.py
> <https://github.com/kaldi-asr/kaldi/pull/706#discussion_r60544896>:
>
> > +                command_opts += "--impulse-response={0} ".format(speech_rir.rir_file_location)
> > +                # add the corresponding isotropic noise if there is any
> > +                if len(speech_rir.iso_noise_list) > 0:
> > +                    isotropic_noise = speech_rir.iso_noise_list[random.randint(0,len(speech_rir.iso_noise_list)-1)]
> > +                    noises_added.append("{0}".format(isotropic_noise.noise_file_location))
> > +                    snrs_added.append("{0}".format(snrs.next()))
> > +                    start_times_added.append(round(random.random() * wav_dur, 2))
> > +
> > +            if noises is not None and random.random() < noise_adding_probability:
> > +                for k in range(random.randint(1, max_noises_added)):
> > +                    # pick the RIR to reverberate the point-source noise
> > +                    noise = noises.next()
> > +                    noise_rir = room['rir_list'][random.randint(0,len(room['rir_list'])-1)]
> > +                    start_times_added.append(round(random.random() * wav_dur, 2))
> > +                    noises_added.append("\"wav-reverberate --duration={2} --impulse-response={1} {0} - |\" ".format(noise.noise_file_location, noise_rir.rir_file_location, round(random.random()*(wav_dur-start_times_added[-1]), 2)))
> > +                    snrs_added.append("{0}".format(snrs.next()))
>
> @vimalmanohar <https://github.com/vimalmanohar> So will isotropic noise
> always be treated as background noise and use the snrs from
> background_snrs? And will isotropic noise always be extended and cover the
> whole file of the speech waveform ?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/kaldi-asr/kaldi/pull/706/files/356fa8258d848e3b092c5b894f6a7b4cab63b29c#r60544896>
>